Fails to compile when installing with ```cargo install ironbar```

**Describe the bug**
When installing through cargo install, everything compiles fine, but when it gets to compile hyprland v0.3.1 , it fails and spits the error down below.

```
error[E0599]: no variant or associated item named `__Nonexhaustive` found for enum `regex::Error` in the current scope
   --> /home/brandon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hyprland-0.3.1/src/event_listener/shared.rs:408:25
    |
408 |         Err(RegexError::__Nonexhaustive) => unreachable!(),
    |                         ^^^^^^^^^^^^^^^ variant or associated item not found in `Error`

error[E0599]: no variant or associated item named `__Nonexhaustive` found for enum `regex::Error` in the current scope
   --> /home/brandon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hyprland-0.3.1/src/event_listener/shared.rs:419:25
    |
419 |         Err(RegexError::__Nonexhaustive) => unreachable!(),
    |                         ^^^^^^^^^^^^^^^ variant or associated item not found in `Error`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `hyprland` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `ironbar v0.12.1`, intermediate artifacts can be found at `/tmp/cargo-installh9aZfY`
``` 
**To Reproduce**
Steps to reproduce the behavior:
1. run cargo install ironbar
2. wait

**Expected behavior**
Complete compilation of hyprland and install. 

Also, after compiling from source, it worked just fine, could be a cargo issue, but I figured I'd ask here just in case.

**System information:**
 - Distro: OpenSUSE Tumbleweed
- Compositor: Hyprland
 - Ironbar version: 0.13.0

**Configuration**

**Additional context**
> Add any other context about the problem here.

**Screenshots**
![image](https://github.com/JakeStanger/ironbar/assets/53587182/740d3960-62ce-47aa-af32-c214c890de9c)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -5,6 +5,7 @@ edition = "2021"
 license = "MIT"
 description = "Customisable GTK Layer Shell wlroots/sway bar"
 repository = "https://github.com/jakestanger/ironbar"
+categories = ["gui"]
 keywords = ["gtk", "bar", "wayland", "wlroots", "gtk-layer-shell"]
 
 [features]
@@ -130,7 +131,7 @@ zbus = { version = "3.14.1", optional = true }
 
 # workspaces
 swayipc-async = { version = "2.0.1", optional = true }
-hyprland = { version = "=0.3.1", optional = true }
+hyprland = { version = "0.3.8", optional = true }
 futures-util = { version = "0.3.21", optional = true }
 
 # shared

--- a/src/clients/compositor/hyprland.rs
+++ b/src/clients/compositor/hyprland.rs
@@ -3,7 +3,7 @@ use crate::{arc_mut, lock, send};
 use color_eyre::Result;
 use hyprland::data::{Workspace as HWorkspace, Workspaces};
 use hyprland::dispatch::{Dispatch, DispatchType, WorkspaceIdentifierWithSpecial};
-use hyprland::event_listener::EventListenerMutable as EventListener;
+use hyprland::event_listener::EventListener;
 use hyprland::prelude::*;
 use hyprland::shared::WorkspaceType;
 use lazy_static::lazy_static;
@@ -46,7 +46,7 @@ impl EventClient {
                 let lock = lock.clone();
                 let active = active.clone();
 
-                event_listener.add_workspace_added_handler(move |workspace_type, _state| {
+                event_listener.add_workspace_added_handler(move |workspace_type| {
                     let _lock = lock!(lock);
                     debug!("Added workspace: {workspace_type:?}");
 
@@ -69,7 +69,7 @@ impl EventClient {
                 let lock = lock.clone();
                 let active = active.clone();
 
-                event_listener.add_workspace_change_handler(move |workspace_type, _state| {
+                event_listener.add_workspace_change_handler(move |workspace_type| {
                     let _lock = lock!(lock);
 
                     let mut prev_workspace = lock!(active);
@@ -105,9 +105,9 @@ impl EventClient {
                 let lock = lock.clone();
                 let active = active.clone();
 
-                event_listener.add_active_monitor_change_handler(move |event_data, _state| {
+                event_listener.add_active_monitor_change_handler(move |event_data| {
                     let _lock = lock!(lock);
-                    let workspace_type = event_data.1;
+                    let workspace_type = event_data.workspace;
 
                     let mut prev_workspace = lock!(active);
 
@@ -134,9 +134,9 @@ impl EventClient {
                 let tx = tx.clone();
                 let lock = lock.clone();
 
-                event_listener.add_workspace_moved_handler(move |event_data, _state| {
+                event_listener.add_workspace_moved_handler(move |event_data| {
                     let _lock = lock!(lock);
-                    let workspace_type = event_data.1;
+                    let workspace_type = event_data.workspace;
                     debug!("Received workspace move: {workspace_type:?}");
 
                     let mut prev_workspace = lock!(active);
@@ -158,7 +158,7 @@ impl EventClient {
             }
 
             {
-                event_listener.add_workspace_destroy_handler(move |workspace_type, _state| {
+                event_listener.add_workspace_destroy_handler(move |workspace_type| {
                     let _lock = lock!(lock);
                     debug!("Received workspace destroy: {workspace_type:?}");
 
@@ -221,9 +221,12 @@ impl EventClient {
 
 impl WorkspaceClient for EventClient {
     fn focus(&self, id: String) -> Result<()> {
-        Dispatch::call(DispatchType::Workspace(
-            WorkspaceIdentifierWithSpecial::Name(&id),
-        ))?;
+        let identifier = match id.parse::<i32>() {
+            Ok(inum) => WorkspaceIdentifierWithSpecial::Id(inum),
+            Err(_) => WorkspaceIdentifierWithSpecial::Name(&id),
+        };
+
+        Dispatch::call(DispatchType::Workspace(identifier))?;
         Ok(())
     }
 

